code/game/objects/structures/grille.dm

```diff
@@ -233,10 +234,15 @@
 		return 0
 	if(!in_range(src, user))//To prevent TK and mech users from getting shocked
 		return 0
+	if(hasShocked)
+		return 0	//Already shocked someone recently?
 	var/turf/T = get_turf(src)
 	var/obj/structure/cable/C = T.get_cable_node()
 	if(C)
 		if(electrocute_mob(user, C, src))
+			hasShocked = 1
+			spawn(10)
```

Comment:
Or maybe even make it store the user it shocks for split second, and then check if the person getting shocked is the stored user?